dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java

```diff
@@ -952,6 +998,7 @@ private void runProcess() {
                 // node success , post node submit
                 if (task.getState() == ExecutionStatus.SUCCESS) {
                     processInstance.setVarPool(task.getVarPool());
+                    processInstance = processService.findProcessInstanceById(processInstance.getId());
```

Comment:
The query db here refreshes the processInstance, The set method of the previous line of code will not take effect.

Reply:
I will fix it ,move "processInstance.setVarPool(task.getVarPool());" to the next line.